test/org/zaproxy/zap/extension/pscanrulesAlpha/UserControlledHTMLAttributesScannerUnitTest.java

```diff
@@ -0,0 +1,213 @@
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfResponseHasNoContentType() {
+		// Given
+		HttpMessage msg = createMessage();
+		msg.setResponseHeader(new HttpResponseHeader());
+		msg.getResponseHeader().setStatusCode(HttpStatusCode.OK);
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfRequestHasNoGetParams() {
+		// Given
+		HttpMessage msg = createMessage();
+		// WHen
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfRequestParamsHaveNoValues() throws URIException, NullPointerException {
+		// Given
+		HttpMessage msg = createMessage();
+		msg.getRequestHeader().setURI(new URI("http://example.com/i.php?place=&name=", false));
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfResponseContainsNoAttributes() throws URIException, NullPointerException {
+		// Given
+		HttpMessage msg = createMessage();
+		msg.getRequestHeader().setURI(new URI("http://example.com/i.php?place=here&name=fred", false));
+		msg.setResponseBody("<html><H1>Title</H1></html>");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfRequestParamValuesNotUsedInAttribute() throws URIException, NullPointerException {
+		// Given
+		HttpMessage msg = createMessage();
+		msg.getRequestHeader().setURI(new URI("http://example.com/i.php?place=here&name=fred", false));
+		msg.setResponseBody("<html><img src=\"x.jpg\" alt=\"Some image (x)\")></img></html>");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfRequestParamValuesNotUsedInMetaAttribute()
+			throws URIException, NullPointerException {
+		// Given
+		HttpMessage msg = createMessage();
+		msg.getRequestHeader().setURI(new URI("http://example.com/i.php?place=here&name=fred", false));
+		msg.setResponseBody("<html><meta name=\"description\" content=\"UnitTest Content\"></html>");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldRaiseAlertIfRequestParamValuesUsedInMetaAttribute() throws URIException, NullPointerException {
+		// Given
+		HttpMessage msg = createMessage();
+		msg.getRequestHeader().setURI(new URI("http://example.com/i.php?place=here&name=fred", false));
+		msg.setResponseBody("<html><meta name=\"description\" content=\"fred\"></html>");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(1));

```

Comment:
Could also assert the alert's param (same for following tests). Worth noting that it will require changing the scanner (currently using hardcoded name "content-type").